driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -393,8 +393,13 @@ private MongoIncompatibleDriverException createIncompatibleException(final Clust
                     MIN_DRIVER_WIRE_VERSION, MIN_DRIVER_SERVER_VERSION);
         } else {
             incompatibleServer = curDescription.findServerIncompatiblyNewerThanDriver();
-            message = format("Server at %s requires wire version %d, but this version of the driver only supports up to %d.",
-                    incompatibleServer.getAddress(), incompatibleServer.getMinWireVersion(), MAX_DRIVER_WIRE_VERSION);
+            if (incompatibleServer != null) {
+                message = format("Server at %s requires wire version %d, but this version of the driver only supports up to %d.",
+                        incompatibleServer.getAddress(), incompatibleServer.getMinWireVersion(), MAX_DRIVER_WIRE_VERSION);
+            } else {
```

Comment:
This conditional will never be entered.  Can we remove it and somehow disable the warning?

Reply:
We could throw an IllegalStateException exception.  Its all juggling for spotbugs that doesn't know if the method call has side effects or not.

Reply:
That seems better to me.